app/src/main/java/com/kickstarter/viewmodels/CommentsViewHolderViewModel.kt

```diff
@@ -163,7 +171,40 @@ interface CommentsViewHolderViewModel {
             comment
                 .compose(takeWhen(this.onRetryViewClicked))
                 .compose(bindToLifecycle())
-                .subscribe(this.retrySendComment)
+                .subscribe {
+                    this.retrySendComment.onNext(it)
+                    this.commentCardStatus.onNext(CommentCardStatus.TRYING_TO_POST)
+                }
+
+            val commentData = this.commentInput.map {
+                Pair(requireNotNull(it.comment?.body()), requireNotNull(it.project))
+            }
+
+            Observable
+                .combineLatest(commentData, onRetryViewClicked) { commentData, _ ->
+                    return@combineLatest this.apolloClient.createComment(
+                        PostCommentData(
+                            project = commentData.second,
+                            body = commentData.first,
+                            clientMutationId = null,
+                            parentId = null
+                        )
+                    ).doOnError {
+                        this.commentCardStatus.onNext(CommentCardStatus.FAILED_TO_SEND_COMMENT)
+                    }
+                        .onErrorResumeNext(Observable.empty())
+                }
+                .switchMap {
+                    it
+                }.subscribe {
+                    this.commentCardStatus.onNext(CommentCardStatus.POSTING_COMMENT_COMPLETED_SUCCESSFULLY)
+                    Handler(Looper.getMainLooper()).postDelayed(
+                        {
+                            this.commentCardStatus.onNext(CommentCardStatus.COMMENT_FOR_LOGIN_BACKED_USERS)
```

Comment:
Quick after thought -> instead of creating a new state, you could emit to both the comment card status as well as the reply button visibility streams. Totally up to you though 😄

Reply:
Actually, this implementation makes sense as the reply button is tied to status anyways, so we can leave as is. Disregard 😅